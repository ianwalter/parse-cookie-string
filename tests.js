const { test } = require('@ianwalter/bff')
const { oneLine, oneLineTrim } = require('common-tags')
const parseCookieString = require('.')

test('simple cookie string gets parsed', t => {
  const cookie = oneLine`
    __stripe_mid=fe67cc7a-cd15-4a30-aa21-260a6207ef55;
    __stripe_sid=cdb94de5-1762-4031-8e98-f8767f442969;
    ${oneLineTrim`
      connect.sid=s%3AY3DYl6RDpmSwt2lGn1FTYuSTLsmBEkk2.
      Sb8i0nDGVPLBGAC8AGCUk6Og8IJzzSFuBzTTYz4jFUY
    `}
  `
  t.expect(parseCookieString(cookie)).toMatchSnapshot()
})
