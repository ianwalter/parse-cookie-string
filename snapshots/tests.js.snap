// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple cookie string gets parsed 1`] = `
Object {
  "__stripe_mid": "fe67cc7a-cd15-4a30-aa21-260a6207ef55",
  "__stripe_sid": "cdb94de5-1762-4031-8e98-f8767f442969",
  "connect.sid": "s%3AY3DYl6RDpmSwt2lGn1FTYuSTLsmBEkk2.Sb8i0nDGVPLBGAC8AGCUk6Og8IJzzSFuBzTTYz4jFUY",
}
`;
